tests/Unit/NumericalAlgorithms/Integration/Test_GslQuad.cpp

```diff
@@ -0,0 +1,90 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cmath>
+#include <cstddef>
+
+#include "NumericalAlgorithms/Integration/GslQuadAdaptive.hpp"
+#include "src/Parallel/Printf.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+// the function in the example
+/// [integrated_function]
+double gaussian(double x, double mean, double factor) {
+  return 2. * factor / sqrt(M_PI) * exp(-square(x - mean));
+}
+/// [integrated_function]
+double integrable_singularity(double x, double factor) {
+  return factor * cos(sqrt(abs(x))) / sqrt(abs(x));
+}
+
+SPECTRE_TEST_CASE("Unit.Numerical.Integration.GslQuadAdaptive",
+                  "[Unit][NumericalAlgorithms]") {
+  double mean = 5.;
+  double factor = 2.0;
+  double lower_bound = -4.;
+  double upper_bound = 10.;
+
+  // Construct the integration and give an example
+  /// [integration_example]
+  integrate::GslQuadAdaptive<integrate::IntegralType::StandardGaussKronrod,
+                             double, double>
+      integration_qag{20};
```

Comment:
Please wrap each "section" of this test in a scope like this:

```cpp
{
  INFO("StandardGaussKronrod");
  // do the test...
}
{
  INFO("InfiniteInterval");
  // next test
}
```

Then you can also name all integration objects just `integration`